miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -53,9 +58,18 @@ export.send_asset
     # is because the VM stack has a minimum size of 16 elements, trying to push
     # elements after the call to `create_note` would increase the stack in
     # addition to the minimum 16 elements.
-    push.0.0 padw movdnw.3 padw movdnw.3 movdn.11 drop
-    # => [ASSET, tag, note_type, RECIPIENT, ZERO, ZERO, 0, ...]
+    push.0 movdn.7 padw movdnw.2 padw movdnw.2
```

Comment:
@hackaugusto can you check if this is still correct?

Reply:
Let's see. 

The goal is to make this work either with `exec.basic::send_asset` or `call.basic::send_asset`, so the stack must look the same after either.

The stack before the `exec.tx::create_note` looks the same for both:

```
# => [tag, aux, note_type, RECIPIENT, x, X, X]
```

The difference starts after `exec.tx::create_note`, since it consumes elements from the stack. When `exec`ed, elements further down the stack move up, and the result looks like this:

```
# => [note_ptr, x, X, X, y, y, Y]
```

When `call`ed, the stack is limited to 16 elements, and instead the VM pads the stack:

```
# => [note_ptr, x, X, X, 0, 0, ZERO]
```

This is the first difference. The second difference, is that when the `call` returns, we don't want `x, X, X` to move up, for the [same reason we have the `syscall`](https://github.com/0xPolygonMiden/miden-base/pull/614#discussion_r1585542386). Instead we want them to stay in the same place:

```
# => [note_ptr, 0, 0, ZERO, x, X, X]
```

This time, the stack further down may be `0, 0, ZERO` for a call, but also `y, y, Y` for an `exec`, so we can't move these values up. Instead we have to manually add the padding and push it down. Before the `exec.tx::create_note` the stack should be:

```
# => [tag, aux, note_type, RECIPIENT, 0, ZERO, ZERO, x, X, X]
```

So that the stack will be the same on either situation:

```
# => [note_ptr, 0, ZERO, ZERO, x, X, X]
```

Looking at the instructions there, it seems correct to me.

Reply:
nice